### ISSUE TITLE
Patitioning feature disabled when using pre-build container from container-registry.oracle.com

### ISSUE THREAD

**Issue by agoretsky** [oracle/docker-images#685]
Hi,
I am running docker container from your registry:
`docker run -d -it --name oracle-db -p 1521:1521 container-registry.oracle.com/database/enterprise:12.2.0.1`

But partitioning feature is disabled there

**Comment by gvenzl** [comment]
Hi @agoretsky, as listed in the [Docker Image Documentation](https://container-registry.oracle.com/pls/apex/f?p=113:4:14151158758980::NO::P4_REPOSITORY,AI_REPOSITORY,AI_REPOSITORY_NAME,P4_REPOSITORY_NAME:9,9,Oracle%20Database%20Enterprise%20Edition,Oracle%20Database%20Enterprise%20Edition&cs=3BhfBdz7F8U_RDUoxsJAzHif9zFlQXxEEjTNL4vWoSq8iW9z6MmoAuqdxNhxZKvEJbKSC2NRF3yoxttjNEL8E1w) Partitioning, nor any other option are supported at this point in time in the prebuilt images:

> Caveats
> This docker image has the following restrictions:
> 
> This Docker image release supports only single database instance.
> Dataguard feature is not supported.
> It does not support enabling of extra options and patching

However, all options and functionality is available in the resulting Docker image from the [Github build files](https://github.com/oracle/docker-images/tree/master/OracleDatabase) here. If you wish to use Partitioning, please use the image resulting from the Dockerfiles instead.

**Comment by veqryn** [comment]
@gvenzl I just ran into this issue when trying to upgrade from `container-registry.oracle.com/database/enterprise:12.1.0.2` to `container-registry.oracle.com/database/enterprise:12.2.0.1`.
Why does 12.1.0.2 support partitioning, but 12.2.0.1 not support it?
This really hurts the usability of this container image for us.  What would it take to reconsider this?

**Comment by gvenzl** [comment]
Hi @veqryn,

That is a very good question. Unfortunately, the 12.2 prebuilt images where designed to be as small as possible, that came with the downside of having functionality removed.
A question here for you and the rest of the community:

Would it be acceptable to have a bigger prebuilt image with about 6GB of size but providing all the functionality?

Thx,

**Comment by veqryn** [comment]
I think so.
You could probably use squash to get it smaller again.

**Comment by gvenzl** [comment]
I use squash today with the GitHub build files and they come down to 6GB. There might be a bit more space savings that can be accomplished but the image size won't go much more down that that. If you still think that this is a valid compromise for having all functionality, please let me know.

**Comment by veqryn** [comment]
Yes, I do

**Comment by veqryn** [comment]
You could have two tags:
12.0.2.1-slim
12.0.2.1-partitions

**Comment by kczulko** [comment]
@gvenzl, I would also agree that having two tags is the best option for the users. They could choose what functionalities do they want, apart from how big the image can unfortunately be.

**Comment by jonheg** [comment]
@gvenzl  Would be nice with an updated images which supports partitioning.
The most basic feature like this should be part of standard Docker images.

**Comment by juergdrollinger** [comment]
What is the state about partitioning. We are missing the feature. On production we need partitioning, so would be nice to use the same SQL scripts in the development on Docker.
A version “12.0.2.1-partition” would be great.

**Comment by jnewton03** [comment]
I also need partitioning in the 12.2 docker container.  Any ETA on that? :)

**Comment by jnewton03** [comment]
@gvenzl I tried using the Github build files, but it takes a really long time to startup since it has to install the software at startup.  We use these containers for immutable environments in our test automation.  Just my use case 2 cents :)

**Comment by gvenzl** [comment]
@jnewton03, it doesn't have to install the software on startup, only once when building the image.
If you like you have can a database as part of the image as well, see https://github.com/oracle/docker-images/tree/master/OracleDatabase/SingleInstance/samples/prebuiltdb

**Comment by jnewton03** [comment]
@gvenzl Thanks for pointing me to that.  I'd like to run a handful of SQL scripts at startup or build time.  Is what you showed me above compatible with this? https://github.com/oracle/docker-images/tree/master/OracleDatabase/SingleInstance#running-scripts-after-setup-and-on-startup

**Comment by shohnn** [comment]
Hi guys, maybe I'm blind (most likely) but exactly which Dockerfile is @gvenzl referring to here?

> However, all options and functionality is available in the resulting Docker image from the [Github build files](https://github.com/oracle/docker-images/tree/master/OracleDatabase) here. If you wish to use Partitioning, please use the image resulting from the Dockerfiles instead.

I've been searching at SingleInstance and RAC dockerfiles but can't seem to find it... 

Also is there any update on adding a --12.0.2.1-partitions tag?

**Comment by rhuffman** [comment]
Why was this issue closed? There's obviously a desire to have partitioning. Everyone on this list is probably in the same boat as me: they use these docker images for testing, and for testing we need partitioning.

The Docker files you provide are mostly great, but they don't actually work. So I'm going to have to spend a lot of time trying to figure out why the install for Oracle 12g fails when building the Docker image, and Oracle installation and Docker are far from my areas of expertise.

**Comment by Djelibeybi** [comment]
If you have any issue with the Dockerfiles provided in this repo, please open a new issue. If you have an issue with the images provided on Oracle Container Registry, please open an SR with Oracle Support.

**Comment by zappee** [comment]
Is there any Oracle 12.2.x Docker image with partitioning feature?
The partition option is also disabled in `store/oracle/database-enterprise:12.2.0.1`  image. Are you planning to create **12.0.2.1-partitions** image?
